bookkeeper-server/src/main/java/org/apache/bookkeeper/util/collections/ConcurrentLongHashMap.java

```diff
@@ -382,6 +382,9 @@ V get(long key, int keyHash) {
         }
 
         V put(long key, V value, int keyHash, boolean onlyIfAbsent, LongFunction<V> valueProvider) {
+            if (value == null) {
+                checkNotNull(valueProvider.apply(key));
```

Comment:
if `valueProvider.apply(key)` returns an object and is not null, it will create at least two objects, this is not expected.

Reply:
yes, I need a variable to store it。 like:
```
        V put(long key, V value, int keyHash, boolean onlyIfAbsent, LongFunction<V> valueProvider) {
            V provideValue;
            if (value == null) {
                provideValue = valueProvider.apply(key);
                if (provideValue == null) {
                    return null;
                }
            }
```
for the compatibility of the program, I feel that returning null directly is better than throwing NullPointerException .
what do you think?

Reply:
There are still problems with this modification, let me think about it